core/time/new_spec.rb

```diff
@@ -475,4 +475,155 @@ def zone.local_to_utc(t)
       end
     end
   end
+
+  ruby_version_is "3.2" do
+    describe "Time.new with a String argument" do
+      it "parses an ISO-8601 like format" do
+        t = Time.utc(2020, 12, 24, 15, 56, 17)
+
+        Time.new("2020-12-24T15:56:17Z").should == t
+        Time.new("2020-12-25 00:56:17 +09:00").should == t
+        Time.new("2020-12-25 00:57:47 +09:01:30").should == t
+        Time.new("2020-12-25 00:56:17 +0900").should == t
+        Time.new("2020-12-25 00:57:47 +090130").should == t
+        Time.new("2020-12-25T00:56:17+09:00").should == t
+      end
+
+      it "accepts precision keyword argument and truncates specified digits of sub-second part" do
+        Time.new("2021-12-25 00:00:00.123456789876 +09:00").subsec.should == 0.123456789r
+        Time.new("2021-12-25 00:00:00.123456789876 +09:00", precision: nil).subsec.should == 0.123456789876r
+        Time.new("2021-12-25 00:00:00 +09:00", precision: 0).subsec.should == 0
+        Time.new("2021-12-25 00:00:00.123456789876 +09:00", precision: -1).subsec.should == Rational(30864197469, 250000000000)
+      end
+
+      it "returns Time in local timezone if not provided in the String argument" do
+        Time.new("2021-12-25 00:00:00").should == Time.new("2021-12-25 00:00:00").getlocal
+      end
```

Comment:
minor: This check is incorrect because `Time#==` will return true for two times in different timezones that point at the same moment (so for instance `#to_i` returns the same Unix timestamp).

On the other hand this test will fail randomly because subsequent calls of `Time.new` will return time with different milliseconds/nanoseconds...

So this check could look like:

```ruby
Time.new("2021-12-25 00:00:00").zone.should == Time.now.zone
Time.new("2021-12-25 00:00:00").utc_offset.should == Time.now.utc_offset
```